### ISSUE TITLE
Add destructuring defaults

### ISSUE THREAD

**Issue by soareschen** [6to5/6to5#230]
Using the latest version 1.14.4, the following code does not work:

```javascript
var obj = { }
var { foo='foo' } = obj
```

Running it throws:

```bash
SyntaxError: destructuring.js: Unexpected token (2:9)
  1  | var obj = { }
> 2  | var { foo='foo' } = obj
```

**Comment by sebmck** [comment]
Destructuring defaults have been added following marijnh/acorn#431a44bc74de4b038f8573e63725688998aef9fd. Support for shorthand `{ foo = "bar" }`) is not supported in the parser yet but `{ foo: foo = "bar" }` is.

**Comment by SimonDegraeve** [comment]
Nice, I was waiting for this for the last couple of months.

**Comment by sebmck** [comment]
Aforementioned destructuring support is available as of 2.6.0

**Comment by sebmck** [comment]
Just so everything's crystal clear the following **will** work:

```javascript
var { a: a = 1 } = {};
a; // 1

function b({ c: c = 2 }){
  c; // 2
}
b({});
```

however the following **will not** until marijnh/acorn#181 is finished:

```javascript
var { a = 1 } = {};
a; // 1

function b({ c = 2 }){
  c; // 2
}
b({});
```

**Comment by sebmck** [comment]
Currently available as of 2.9.0

**Comment by fkling** [comment]
Awesome!
